ecc_enclave/enclave/shim.cpp

```diff
@@ -92,7 +94,41 @@ void get_state(
     if (ret != SGX_SUCCESS)
     {
         LOG_ERROR("Enclave: Error decrypting state: %d", ret);
-        *val_len = 0;
+        throw std::runtime_error("Enclave: Error decrypting state");
+    }
+}
+
+void get_public_state(
```

Comment:
My main comment is related to code duplication in `put/get_state` and `put/get_public_state` for retrieving and checking data. These could be consolidated by making the `public` version compute cmac's and do ocalls, and by implementing the `private` version based on the public one (plus data encryption/decryption).

Reply:
makes sense.